database/storage.go

```diff
@@ -137,9 +126,54 @@ func (s *Storage) rows(table string) (*sql.Rows, error) {
 }
 
 func (s *Storage) nRows(table string, n string) (*sql.Rows, error) {
-	nRows, err := s.conn.Query(fmt.Sprintf("SELECT * FROM `%s` LIMIT %s", table, n))
+	column, err := s.primaryColumn(table)
+	if err != nil {
+		return nil, err
+	}
+	nRows, err := s.conn.Query(fmt.Sprintf("SELECT * FROM `%s` ORDER BY %s DESC LIMIT %s", table, column, n))
 	if err != nil {
 		return nRows, err
 	}
 	return nRows, nil
 }
+
+// database returns the name of the database.
+func (s *Storage) database() (string, error) {
+	var db string
```

Comment:
I would recommend to introduce the variable near the place you would actually use it.